pkg/modelzooserver/modelzooserver.go

```diff
@@ -15,28 +15,162 @@ package modelzooserver
 
 import (
 	"context"
+	"fmt"
+	"io"
+	"log"
 
+	"sqlflow.org/sqlflow/pkg/database"
 	pb "sqlflow.org/sqlflow/pkg/proto"
 )
 
+const modelCollTable = "sqlflow_model_zoo.model_collection"
+const modelDefTable = "sqlflow_model_zoo.model_definition"
+const trainedModelTable = "sqlflow_model_zoo.trained_model"
+
+// TODO(typhoonzero): create tables if these tables are not pre created?
+const createTableStmts = `CREATE DATABASE IF NOT EXISTS sqlflow_model_zoo;
+DROP TABLE IF EXISTS sqlflow_model_zoo.trained_model;
+DROP TABLE IF EXISTS sqlflow_model_zoo.model_definition;
+DROP TABLE IF EXISTS sqlflow_model_zoo.model_collection;
+
+CREATE TABLE sqlflow_model_zoo.model_collection (
+    id INT AUTO_INCREMENT,
+    name VARCHAR(255),
+    version VARCHAR(255),
+    PRIMARY KEY (id)
+);
+
+CREATE TABLE sqlflow_model_zoo.model_definition (
+    id INT AUTO_INCREMENT,
+    model_coll_id INT,
+    class_name VARCHAR(255),
+    args_desc VARCHAR(255),
+    PRIMARY KEY (id),
+    FOREIGN KEY (model_coll_id) REFERENCES model_collection(id)
+);
+
+CREATE TABLE sqlflow_model_zoo.trained_model (
+    id INT AUTO_INCREMENT,
+    model_def_id INT,
+    name VARCHAR(255),
+    version VARCHAR(255),
+    url VARCHAR(255),
+    description TEXT,
+    metrics TEXT,
+    PRIMARY KEY (id),
+    FOREIGN KEY (model_def_id) REFERENCES model_definition(id)
+);`
+
 type modelZooServer struct {
+	DB *database.DB
 }
 
 func (s *modelZooServer) ListModelDefs(ctx context.Context, req *pb.ListModelRequest) (*pb.ListModelResponse, error) {
```

Comment:
I took a look into the proto file.   It seems that the size field in ListModelReponse is redundant to `len(names)`.

```protobuf
message ListModelResponse {
    // A list of model definition or trained model names.
    repeated string names = 1;
    // A list of tags of the above list names, the length should be the same as "names".
    repeated string tags = 2;
    // Size of elements in names.
    int64 size = 3;
}
```

Reply:
How if users want to list model definitions belonging to a specific model collection/repo?  I don't see a field in `ListModelRequest` allowing users to specify the filtering condition.

I would suggest that we follow the convention of GitHub API to have

```protobuf
message /*model_zoo.*/ListRepoRequest {...}
message /*model_zoo.*/ListRepoResponse {...}

message /*model_zoo.*/ListDefinitionRequest { string repo; ...}
message /*model_zoo.*/ListDefinitionResponse {...}

message /*model_zoo.*/ListModelRequest { string repo; string definition; ...}
message /*model_zoo.*/ListModelResponse {...}
```

Reply:
Planning to add this in the next PR.